Tests/NIOHTTPCompressionTests/HTTPRequestDecompressorTest.swift

```diff
@@ -65,12 +65,10 @@ class HTTPRequestDecompressorTest: XCTestCase {
 
         do {
```

Comment:
Instead of this do catch block I was trying `XCTAssertThrowsSpecific` but it was not working: 

```
error: use of unresolved identifier 'XCTAssertThrowsSpecific'
```

Reply:
That function is for exceptions only: you should use `XCTAssertThrowsError`.

Reply:
Actually I was getting error with AssetEqual: ` error: generic parameter 'T' could not be inferred`

Reply:
`DecompressionError` is not `Equatable`, so that could not work. It's unrelated to the use of `XCTAssertThrowsError` though: it only affects the code in the `error` block.

Reply:
So, should I revert  the related changes (`if case .limit = `) in all the file ?

Reply:
Those changes are fine, but they need to be moved into the closure that `XCTAssertThrowsError` provides.

Reply:
I think last time I tested there were error : 

```
HTTPResponseDecompressorTest.swift:91: error: -[NIOHTTPCompressionTests.HTTPResponseDecompressorTest testDecompression] : XCTAssertThrowsError failed: did not throw an error
```